### PR TITLE
Fixes error prone String#split warnings

### LINKED ISSUE #1823: Fix String.split() warnings by errorprone
Errorprone reports warnings: Guava Splitter is preferred to String.split().
 (see http://errorprone.info/bugpattern/StringSplitter)

### PR DESCRIPTION
**Pull request by etanshaul**
fixes #1823

we were getting these warnings on our build:
http://errorprone.info/bugpattern/StringSplitter

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/application/GoogleApiClientAppEngineAdminService.java [hunk 1]
```diff
@@ -159,7 +161,7 @@ private Operation getOperation(
       @NotNull String projectId, @NotNull String operationName, @NotNull Credential credential)
       throws IOException {
     // The operation ID is the final slash-separated component of the operation name.
-    String[] nameParts = operationName.split("/");
+    String[] nameParts = Iterables.toArray(Splitter.on("/").split(operationName), String.class);
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/application/GoogleApiClientAppEngineAdminService.java]: was trying to make the least invasive changes here to quiet the warning. if you think just adding `@SuppressWarnings("StringSplitter")` is better let me know.

**Comment by ivanporty** [comment]
Nice, I noticed that too and created an issue for that just in the same time!

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/application/GoogleApiClientAppEngineAdminService.java (hunk 1 above)
Comment (ivanporty) [reply google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/application/GoogleApiClientAppEngineAdminService.java]: Maybe use `splitToList`? Kind of unusual to see String.class. It looks like List would work too

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/ServerToIdeFileResolver.java [hunk 2]
```diff
@@ -110,7 +112,7 @@ public VirtualFile getFileFromPath(@NotNull Project project, @NotNull String pat
    */
   @VisibleForTesting
   static String getPackageFromPath(String path) {
-    String[] tokens = path.split("/");
+    String[] tokens = Iterables.toArray(Splitter.on("/").split(path), String.class);
```
Comment (ivanporty) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/ServerToIdeFileResolver.java]: ditto (`splitToList`)?

**Comment by etanshaul** [comment]
cool. updated to use splitToList. Let me know if thats not what you had in mind.

**Pull request merged**